plugins/inputs/exec/exec.go

```diff
@@ -63,7 +64,12 @@ func (c CommandRunner) Run(e *Exec, command string) ([]byte, error) {
 	cmd.Stdout = &out
 
 	if err := cmd.Run(); err != nil {
-		return nil, fmt.Errorf("exec: %s for command '%s'", err, command)
+		switch e.parser.(type) {
+		case *nagios.NagiosParser:
```

Comment:
Because nagios plugins use exit code to indicate the check status
* 0 == OK
* 1 == WARNING
* 3 == CRITICAL
* 4 or more == UNKNOWN
Maybe I should add a state metric for it ?